scrape/manager.go

```diff
@@ -190,6 +192,36 @@ func (m *Manager) reload() {
 	}
 	m.mtxScrape.Unlock()
 	wg.Wait()
+
+	m.warnIfTargetsRelabelledToSameLabels()
+}
+
+func (m *Manager) warnIfTargetsRelabelledToSameLabels() {
+	m.mtxScrape.Lock()
+	defer m.mtxScrape.Unlock()
+
+	totalTargets := 0
+	for _, scrapePool := range m.scrapePools {
+		totalTargets += len(scrapePool.activeTargets)
+	}
+
+	activeTargets := make(map[uint64]*Target, totalTargets)
+	for _, scrapePool := range m.scrapePools {
+		for _, target := range scrapePool.activeTargets {
+			lHash := target.labels.Hash()
+			t, ok := activeTargets[lHash]
```

Comment:
There is some risk that two sets of labels will hash to the same value; it would be safer to make the map key `labels.Bytes()`.